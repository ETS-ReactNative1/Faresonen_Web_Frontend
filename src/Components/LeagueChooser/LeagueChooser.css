.disabled {
    opacity: 0.5
}

a.ui.card.disabled:hover {
    transform: none;
    cursor: not-allowed;
}