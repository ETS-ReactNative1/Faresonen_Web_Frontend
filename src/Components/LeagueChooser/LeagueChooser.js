import React from "react";
import PropTypes from "prop-types";
import { Grid, Card, Message } from "semantic-ui-react";

const LeagueChooser = props => {
  const leagues = props.leagues.map(league => {
    const activeSeason = league.activeseason["@uri"].match(/\d+/gm)[1];
    return (
      <Grid.Column key={league.id}>
        <Card
          href={`#/league/${league.name}/${league.id}/${activeSeason}/`}
          header={league.name}
          description={`TurneringsID: ${league.id} - SesongID: ${activeSeason}`}
        />
      </Grid.Column>
    );
  });
  return leagues.length > 0 ? (
    <Grid columns={3}>{leagues}</Grid>
  ) : (
    <Message info>
      <Message.Header>Ingen ligaer tilgjengelig</Message.Header>
      <p>
        Det er ingen ligaer lagt til i databasen. Dersom du er administrator kan
        du legge til kamper <a href="#/admin">her</a> eller få en administrator
        til å gjøre det for deg.
      </p>
    </Message>
  );
};

LeagueChooser.propTypes = {
  leagues: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      activeseason: PropTypes.shape({
        "@uri": PropTypes.string.isRequired
      })
    })
  ).isRequired
};

export default LeagueChooser;
