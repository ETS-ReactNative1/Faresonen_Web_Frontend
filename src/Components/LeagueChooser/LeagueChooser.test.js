import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import LeagueChooser from "./LeagueChooser";
import data from "../../Data/leagues";

describe("LeagueChooser", () => {
  it("Should render properly", () => {
    const tree = renderer.create(<LeagueChooser leagues={data.leagues} />);
    expect(tree).toMatchSnapshot();
  });
  const elem = shallow(<LeagueChooser leagues={data.leagues} />);
  describe("League props should have data about leagues", () => {
    it("should be rendered a child component for each league defined", () => {
      expect(elem.props().children.length).toEqual(data.leagues.length);
    });
  });
});
