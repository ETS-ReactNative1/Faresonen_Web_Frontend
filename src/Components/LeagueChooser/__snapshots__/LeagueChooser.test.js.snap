// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeagueChooser Should render properly 1`] = `
<div
  className="ui three column grid"
>
  <div
    className="column"
  >
    <a
      className="ui card"
      href="#/league/Eliteserien/1/340/"
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="header"
        >
          Eliteserien
        </div>
        <div
          className="description"
        >
          TurneringsID: 1 - SesongID: 340
        </div>
      </div>
    </a>
  </div>
  <div
    className="column"
  >
    <a
      className="ui card"
      href="#/league/OBOS-ligaen/2/340/"
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="header"
        >
          OBOS-ligaen
        </div>
        <div
          className="description"
        >
          TurneringsID: 2 - SesongID: 340
        </div>
      </div>
    </a>
  </div>
  <div
    className="column"
  >
    <a
      className="ui card"
      href="#/league/Premier League/230/339/"
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="header"
        >
          Premier League
        </div>
        <div
          className="description"
        >
          TurneringsID: 230 - SesongID: 339
        </div>
      </div>
    </a>
  </div>
</div>
`;
